From db72479eea4b5bbd73d1fa33f4c9e4e4e2b99b04 Mon Sep 17 00:00:00 2001
From: Nick Mathewson <nickm@torproject.org>
Date: Mon, 10 Mar 2014 15:01:27 -0400
Subject: [PATCH] Update ns downloads when we receive a bridge descriptor

This prevents long stalls when we're starting with a state file but
with no bridge descriptors.  Fixes bug 9229.  I believe this bug has
been present since 0.2.0.3-alpha.
---
 changes/bug9229     | 5 +++++
 src/or/entrynodes.c | 6 +++++-
 2 files changed, 10 insertions(+), 1 deletion(-)
 create mode 100644 changes/bug9229

diff --git a/changes/bug9229 b/changes/bug9229
new file mode 100644
index 0000000..ad7fd22
--- /dev/null
+++ b/changes/bug9229
@@ -0,0 +1,5 @@
+  o Minor bugfixes:
+    - Avoid 60-second delays in the bootstrapping process when Tor
+      is launching for a second time while using bridges. Fixes bug 9229;
+      bugfix on 0.2.0.3-alpha.
+
diff --git a/src/or/entrynodes.c b/src/or/entrynodes.c
index 2aa063c..59770fa 100644
--- a/src/or/entrynodes.c
+++ b/src/or/entrynodes.c
@@ -2115,8 +2115,12 @@ learned_bridge_descriptor(routerinfo_t *ri, int from_cache)
        * our entry node list */
       entry_guard_register_connect_status(ri->cache_info.identity_digest,
                                           1, 0, now);
-      if (first)
+      if (first) {
+        /* XXXX apparently, this is never called. See bug #9229. */
         routerlist_retry_directory_downloads(now);
+      }
+
+      update_networkstatus_downloads(now);
     }
   }
 }
-- 
1.8.1.2

